src/Microsoft.Restier.WebApi/Query/QueryExecutorWebApi.cs

```diff
@@ -0,0 +1,45 @@
+﻿// Copyright (c) Microsoft Corporation.  All rights reserved.
+// Licensed under the MIT License.  See License.txt in the project root for license information.
+
+using System.Linq;
+using System.Linq.Expressions;
+using System.Threading;
+using System.Threading.Tasks;
+using System.Web.OData.Extensions;
+using Microsoft.Restier.Core.Query;
+
+namespace Microsoft.Restier.WebApi.Query
+{
+    internal class QueryExecutorWebApi : IQueryExecutor
+    {
+        public IQueryExecutor Inner { get; set; }
+
+        public async Task<QueryResult> ExecuteQueryAsync<TElement>(
+            QueryContext context,
+            IQueryable<TElement> query,
+            CancellationToken cancellationToken)
+        {
+            var queryContext = context.ApiContext.GetApiService<WebApiContext>();
+            if (queryContext.QueryIncludeTotalCount == true)
+            {
+                var countQuery = ExpressionHelpers.GetCountableQuery(query);
+                var expression = countQuery.Expression;
+                expression = ExpressionHelpers.Count(expression, countQuery.ElementType);
```

Comment:
What's the major purpose of this call? seems we make it generic with specific type, and make it as call, the expression is already object type which is enough for count, also I have not tested whether countQuery.ElementType can get more specific type here besides object.